hooks/browser_integration.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+class BrowserIntegration(HookBaseClass):
+    def get_cache_key(self, config_uri, project, entity_type):
```

Comment:
We handle shot vs. asset tasks already. The data required to make that determination is not passed down from SG, however; it's queried and cached (both in the web app and in Python).